[CI] UpgradeClusterClientYamlTestSuiteIT failure in suite :x-pack:qa:rolling-upgrade:v7.6.0#twoThirdsUpgradedTest

**Build scan**: https://gradle-enterprise.elastic.co/s/xazhuw2lv4xok

**Repro line**:
```
gradlew ':x-pack:qa:rolling-upgrade:v7.6.0#twoThirdsUpgradedTest' -Dtests.class="org.elasticsearch.upgrades.UpgradeClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=mixed_cluster/80_transform_jobs_crud/Test GET, start, and stop old cluster batch transforms}" -Dtests.seed=B919838B66982C1D -Dtests.bwc=true -Dtests.locale=zh-HK -Dtests.timezone=Africa/Conakry -Druntime.java=8
```

**Reproduces locally?**:  yes, but the assertion error provided below is not in the nodes' logs

**Applicable branches**: 7.x

**Failure history**:
There were some failures starting yesterday, but the assertion failure is not there.

**Failure excerpt**:

```
09:23:51 » [2021-07-15T06:16:09,351][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [v7.6.0-0] fatal error in thread [Thread-7], exiting
09:23:51 »  java.lang.AssertionError: org.elasticsearch.search.internal.ShardSearchRequest/eJ60j0jESEa9UZLXUQcC2w:511
09:23:51 »  	at org.elasticsearch.search.SearchService.getShard(SearchService.java:431) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
09:23:51 »  	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:405) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
09:23:51 »  	at org.elasticsearch.action.search.SearchTransportService.lambda$registerRequestHandler$6(SearchTransportService.java:311) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
09:23:51 »  	at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler$1.doRun(SecurityServerTransportInterceptor.java:261) ~[?:?]
09:23:51 »  	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
09:23:51 »  	at org.elasticsearch.common.util.concurrent.EsExecutors$DirectExecutorService.execute(EsExecutors.java:228) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
09:23:51 »  	at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.lambda$messageReceived$0(SecurityServerTransportInterceptor.java:315) ~[?:?]
09:23:51 »  	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:134) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
```

It seems there are some incompatibilities between 7.x and 7.6 with the above assertion failing.
Some tests are failing because the test suite times out, and some fail because one of the nodes is not available, but the assertion error above looks troubling nonetheless.

--- a/qa/mixed-cluster/src/test/java/org/elasticsearch/backwards/SearchWithMinCompatibleSearchNodeIT.java
+++ b/qa/mixed-cluster/src/test/java/org/elasticsearch/backwards/SearchWithMinCompatibleSearchNodeIT.java
@@ -142,4 +142,21 @@ public class SearchWithMinCompatibleSearchNodeIT extends ESRestTestCase {
             }
         }
     }
+
+    public void testPointInTimeRequirement() throws Exception {
+        final Request openPIT = new Request("POST", index + "/_pit");
+        openPIT.addParameter("keep_alive", "10m");
+        if (nodes.getBWCNodes().stream().anyMatch(node -> node.getVersion().before(Version.V_7_10_0))) {
+            ResponseException responseException = expectThrows(ResponseException.class, () -> client().performRequest(openPIT));
+            assertThat(responseException.getResponse().getStatusLine().getStatusCode(), equalTo(400));
+            assertThat(responseException.getMessage(), containsString("Point-in-time requires every node in the cluster on 7.10 or later"));
+        } else {
+            final Response response = client().performRequest(openPIT);
+            assertOK(response);
+            final String pitID = ObjectPath.createFromResponse(response).evaluate("id");
+            final Request closePIT = new Request("DELETE", "_pit");
+            closePIT.setJsonEntity("{\"id\":\"" + pitID + "\"}");
+            assertOK(client().performRequest(closePIT));
+        }
+    }
 }

--- a/qa/mixed-cluster/src/test/java/org/elasticsearch/backwards/SearchWithMinCompatibleSearchNodeIT.java
+++ b/qa/mixed-cluster/src/test/java/org/elasticsearch/backwards/SearchWithMinCompatibleSearchNodeIT.java
@@ -29,6 +29,7 @@ import java.util.List;
 import java.util.Map;
 
 import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.either;
 import static org.hamcrest.Matchers.equalTo;
 
 public class SearchWithMinCompatibleSearchNodeIT extends ESRestTestCase {
@@ -149,7 +150,9 @@ public class SearchWithMinCompatibleSearchNodeIT extends ESRestTestCase {
         if (nodes.getBWCNodes().stream().anyMatch(node -> node.getVersion().before(Version.V_7_10_0))) {
             ResponseException responseException = expectThrows(ResponseException.class, () -> client().performRequest(openPIT));
             assertThat(responseException.getResponse().getStatusLine().getStatusCode(), equalTo(400));
-            assertThat(responseException.getMessage(), containsString("Point-in-time requires every node in the cluster on 7.10 or later"));
+            assertThat(responseException.getMessage(),
+                either(containsString("request body is required"))
+                    .or(containsString("Point-in-time requires every node in the cluster on 7.10 or later")));
         } else {
             final Response response = client().performRequest(openPIT);
             assertOK(response);

--- a/server/src/main/java/org/elasticsearch/action/search/TransportOpenPointInTimeAction.java
+++ b/server/src/main/java/org/elasticsearch/action/search/TransportOpenPointInTimeAction.java
@@ -8,6 +8,7 @@
 
 package org.elasticsearch.action.search;
 
+import org.elasticsearch.Version;
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.action.ActionListenerResponseHandler;
 import org.elasticsearch.action.IndicesRequest;
@@ -16,6 +17,7 @@ import org.elasticsearch.action.support.ActionFilters;
 import org.elasticsearch.action.support.ChannelActionListener;
 import org.elasticsearch.action.support.HandledTransportAction;
 import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.cluster.service.ClusterService;
 import org.elasticsearch.common.inject.Inject;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
@@ -40,18 +42,21 @@ public class TransportOpenPointInTimeAction extends HandledTransportAction<OpenP
     private final TransportSearchAction transportSearchAction;
     private final TransportService transportService;
     private final SearchService searchService;
+    private final ClusterService clusterService;
 
     @Inject
     public TransportOpenPointInTimeAction(
         TransportService transportService,
         SearchService searchService,
         ActionFilters actionFilters,
+        ClusterService clusterService,
         TransportSearchAction transportSearchAction
     ) {
         super(OpenPointInTimeAction.NAME, transportService, actionFilters, OpenPointInTimeRequest::new);
         this.transportService = transportService;
         this.transportSearchAction = transportSearchAction;
         this.searchService = searchService;
+        this.clusterService = clusterService;
         transportService.registerRequestHandler(
             OPEN_SHARD_READER_CONTEXT_NAME,
             ThreadPool.Names.SAME,
@@ -68,6 +73,13 @@ public class TransportOpenPointInTimeAction extends HandledTransportAction<OpenP
 
     @Override
     protected void doExecute(Task task, OpenPointInTimeRequest request, ActionListener<OpenPointInTimeResponse> listener) {
+        final Version minNodeVersion = clusterService.state().nodes().getMinNodeVersion();
+        if (minNodeVersion.before(Version.V_7_10_0)) {
+            listener.onFailure(
+                new IllegalArgumentException("Point-in-time requires every node in the cluster on 7.10 or later; " +
+                    "got [" + minNodeVersion + "]"));
+            return;
+        }
         final SearchRequest searchRequest = new SearchRequest().indices(request.indices())
             .indicesOptions(request.indicesOptions())
             .preference(request.preference())

